File: Tier1/Shared/DTOs/HousingDTO.cs
using Shared.Domain;

namespace Shared.DTOs;

public class HousingDTO
{
    public Housing Housing{ get; }
    public string ErrorMessage{ get; }

    public HousingDTO(Housing housing, string errorMessage)
    {
        Housing = housing;
        ErrorMessage = errorMessage;
    }
}

File: Tier2/src/main/java/via/sep3/group11/tier2/shared/DTOs/HousingDTO.java
package via.sep3.group11.tier2.shared.DTOs;

import via.sep3.group11.tier2.shared.domain.Housing;

public class HousingDTO {
    private Housing housing;
    private String errorMessage;

    public HousingDTO(Housing housing, String errorMessage) {
        this.housing = housing;
        this.errorMessage = errorMessage;
    }

    public Housing getHousing() {
        return housing;
    }

    public String getErrorMessage() {
        return errorMessage;
    }
}
